apps/test/unit/dance/ai/utils/chooseEffectsTest.js

```diff
@@ -0,0 +1,46 @@
+import {expect} from '../../../../util/reconfiguredChai';
+import sinon from 'sinon';
+import {chooseEffects} from '@cdo/apps/dance/ai/utils/chooseEffects';
+import * as calculateOutputSummedWeights from '@cdo/apps/dance/ai/utils/calculateOutputSummedWeights';
+import {EffectsQuality} from '@cdo/apps/dance/ai/types';
+
+describe('chooseEffects', () => {
+  beforeEach(() => {
+    sinon.stub(Math, 'random').returns(0.99);
+    sinon
+      .stub(calculateOutputSummedWeights, 'calculateOutputSummedWeights')
+      .returns([
+        [2.9, 'blooming_petals (1st)'],
+        [1.2, 'quads (3rd)'],
+        [0.3, 'circles (4th)'],
+        [0.1, 'clouds (5th)'],
+        [2.1, 'color_cycle (2nd)'],
+      ]);
+  });
+
+  afterEach(() => {
+    sinon.restore();
+  });
+
+  it('chooses a good effect', () => {
+    const chosenEffects = chooseEffects(
+      ['cyclone', 'smiling-face-with-hearts', 'party-popper'],
+      EffectsQuality.GOOD
+    );
+
```

Comment:
Nice test! What do you think of adding a brief comment on why the expected value is the 3rd option? I did a little double-take at first glance but then looked at the code (`NUM_RANDOM_TOP_OPTIONS = 3` and stubbing of `Math.random`).